Specs/Scene/MaterialSpec.js

```diff
@@ -655,6 +655,206 @@ describe(
         });
     });
 
+    it("handles when material image is undefined", function () {
+      var material = Material.fromType(Material.ImageType, {
+        image: undefined,
+        color: Color.RED,
+      });
+      renderMaterial(material, false, function (rgba) {
+        expect(rgba).toEqual([255, 0, 0, 255]);
+      });
+    });
+
+    it("handles when material image is set to default image", function () {
+      var material = Material.fromType(Material.ImageType, {
+        image: Material.DefaultImageId,
+        color: Color.RED,
+      });
+      renderMaterial(material, false, function (rgba) {
+        expect(rgba).toEqual([255, 0, 0, 255]);
+      });
+    });
+
+    it("handles when material image is changed from undefined to some image", function () {
+      var material = Material.fromType(Material.ImageType, {
+        image: undefined,
+        color: Color.WHITE,
+      });
+
+      renderMaterial(material, false, function (rgba) {
+        expect(rgba).toEqual([255, 255, 255, 255]);
+      });
+
+      material.uniforms.image = "./Data/Images/Green.png";
+      return pollToPromise(
+        function () {
+          renderMaterial(material, true);
+          return material._textures["image"] !== material._defaultTexture;
+        },
+        { timeout: 10000 }
+      ).then(function () {
+        renderMaterial(material, true, function (rgba) {
+          expect(rgba).toEqual([0, 255, 0, 255]);
+        });
+      });
+    });
+
+    it("handles when material image is changed from default to some image", function () {
+      var material = Material.fromType(Material.ImageType, {
+        image: Material.DefaultImageId,
+        color: Color.WHITE,
+      });
+
+      renderMaterial(material, false, function (rgba) {
+        expect(rgba).toEqual([255, 255, 255, 255]);
+      });
+
+      material.uniforms.image = "./Data/Images/Green.png";
+      return pollToPromise(
+        function () {
+          renderMaterial(material, true);
+          return material._textures["image"] !== material._defaultTexture;
+        },
+        { timeout: 10000 }
```

Comment:
I don't think the timeouts are necessary. These images should be loaded well before the default timeout.